src/lib.rs

```diff
@@ -40,15 +40,22 @@
 
 extern crate libc;
 
-use libc::{pid_t, c_char, c_void, getpid, size_t, c_int};
+use libc::{pid_t, c_char, c_void, getpid, size_t};
 use std::ffi::CString;
 use std::path::PathBuf;
 
+#[allow(dead_code)]
```

Comment:
Why do we need this? Isn't it also a module level directive, not just attached to an `enum`?

Reply:
it's because we don't use `CFalse`, but we kinda do want it's definition.

```
warning: variant is never constructed: `CFalse`
  --> src/lib.rs:50:5
   |
50 |     CFalse = 0,
   |     ^^^^^^^^^^
   |
   = note: #[warn(dead_code)] on by default
```

Reply:
Is the directive per-enum or per-module?

Reply:
it's just for this enum. If I had put it right at the top of the file and prepended with a `!` it would have been for the whole module.

```
#[allow(dead_code)]    
#[repr(u8)]    
enum CBool {    
    CFalse = 0,    
    CTrue = 1,    
}    
    
enum J {    
    X = 1,    
}    
```

```
vext01@bencher8:~/research/traceme$ cargo build --examples
   Compiling traceme v0.1.0 (file:///home/vext01/research/traceme)
warning: enum is never used: `J`
  --> src/lib.rs:54:1
   |
54 | enum J {
   | ^^^^^^
   |
   = note: #[warn(dead_code)] on by default
```

Reply:
OK, I did not know one could do that.